core/admin/port_management.py

```
@@ -0,0 +1,449 @@
+            "iptables", "-A", "INPUT",
+            "-p", "tcp",
+            "--dport", str(external_port),
+            "-j", "ACCEPT"
+        ], check=True, capture_output=True)
+        
+        logger.info(f"Added iptables INPUT rule for port {external_port}")
+    except subprocess.CalledProcessError as e:
+        logger.error(f"Failed to add iptables INPUT rule: {e}")
+        raise ValueError(f"Failed to open port: {e}")
+    
+    now = datetime.now()
```

Ruff (0.14.10)

`datetime.datetime.now()` called without a `tz` argument

(DTZ005)